sdk/search/Azure.Search.Documents/src/Models/SearchResults{T}.cs

```diff
@@ -231,7 +229,12 @@ public SearchAsyncPageable(SearchResults<T> results)
         /// <inheritdoc />
         public override async IAsyncEnumerable<Page<SearchResult<T>>> AsPages(string continuationToken = default, int? pageSizeHint = default)
         {
-            SearchResults<T> initial = _results; // TODO: #10590 - Add durable continuation tokens
+            // The first page of our results is always provided so we can
+            // ignore the continuation token.  Users can only provide a token
+            // directly to the Search method.
+            Debug.Assert(continuationToken == null);
```

Comment:
Given the comment, are we not planning on testing this eventually then, for which this will assert?

Reply:
There's not really anything to test.  We just ignore the value.  I added a `Debug.Assert` as a friendly reminder that somebody screwed something up if this happens while we're working on it.  We could consider strengthening that to an exception but I kind of want Krzysztof to give this whole area another look anyway.   We made the call to squish Pageable behind a Response and this is a consequence.  I think it's okay, but more eyes before GA will be good.